typescript/test/data/redemption.ts

```diff
@@ -0,0 +1,484 @@
+import { BigNumber, BytesLike } from "ethers"
+import { RawTransaction, UnspentTransactionOutput } from "../../src/bitcoin"
+import { RedemptionRequest } from "../../src/redemption"
+
+/**
+ * Private key (testnet) of the wallet.
+ */
+export const walletPrivateKey =
+  "cRk1zdau3jp2X3XsrRKDdviYLuC32fHfyU186wLBEbZWx4uQWW3v"
+
+/**
+ * P2PKH address corresponding to walletPrivateKey.
+ */
+export const p2pkhWalletAddress = "mtSEUCE7G8om9zJttG9twtjoiSsUz7QnY9"
+
+/**
+ * P2WPKH address corresponding to walletPrivateKey.
+ */
+export const p2wpkhWalletAddress = "tb1q3k6sadfqv04fmx9naty3fzdfpaecnphkfm3cf3"
+
+/**
+ * Represents a set of data used for given sweep scenario.
+ */
+export interface RedemptionTestData {
+  mainUtxo: UnspentTransactionOutput & RawTransaction
+  pendingRedemptions: {
+    redemptionKey: BytesLike
+    pendingRedemption: RedemptionRequest
+  }[]
+  witness: boolean
+  expectedRedemption: {
+    transactionHash: string
+    transaction: RawTransaction
+  }
+}
+
+/**
+ * Test data that is based on a Bitcoin redemption transaction with a single
+ * P2PKH redeemer address:
+ * https://live.blockcypher.com/btc-testnet/tx/c437f1117db977682334b53a71fbe63a42aab42f6e0976c35b69977f86308c20/
+ */
+export const singleP2PKHAddressRedemption: RedemptionTestData = {
+  mainUtxo: {
+    transactionHash:
+      "523e4bfb71804e5ed3b76c8933d733339563e560311c1bf835934ee7aae5db20",
+    outputIndex: 1,
+    value: 1481680,
+    transactionHex:
+      "0100000000010160d264b34e51e6567254bcaf4cc67e1e069483f4249dc50784eae68" +
+      "2645fd11d0100000000ffffffff02d84000000000000022002086a303cdd2e2eab1d1" +
+      "679f1a813835dc5a1b65321077cdccaf08f98cbf04ca96d09b1600000000001600148" +
+      "db50eb52063ea9d98b3eac91489a90f738986f602483045022100ed5fa06ea5e9d4a9" +
+      "f0cf0df86a2cd473f693e5bda3d808ba82b04ee26d72b73f0220648f4d7bb25be7819" +
+      "22349d382cf0f32ffcbbf89c483776472c2d15644a48d67012103989d253b17a6a0f4" +
+      "1838b84ff0d20e8898f9d7b1a98f2564da4cc29dcf8581d900000000",
+  },
+  pendingRedemptions: [
+    {
+      redemptionKey:
+        "0xe7a0e8b724c3ee7708701137cb57f0a742e284b4ad0b21cc16a3b3fbe733521a",
```

Comment:
Can we use un-prefixed `redemptionKey` across this dataset?

Reply:
As discussed, we decided to keep the prefix, as the result of `solidityKeccak256` is prefixed.